### PR TITLE
store/tikv: increase max backoff time of read requests

### PR DESCRIPTION
**Pull request by sticnarf**
### What is changed and how it works?

Like how https://github.com/pingcap/tidb/pull/25040 changes prewrite and commit, this PR increases the max backoff to 10 minutes. So, if some TiKV regions are unavailable in a short time due to slow apply, the user will not receive an error.

This PR also make real `EpochNotMatch` errors not backoff.

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

### Release note 

- `Increase timeout for read requests to avoid propagating TiKV errors.`

### CODE REVIEW

File: store/tikv/2pc.go [hunk 1]
```diff
@@ -745,7 +746,7 @@ func (tm *ttlManager) keepAlive(c *twoPhaseCommitter) {
 			if tm.lockCtx != nil && tm.lockCtx.Killed != nil && atomic.LoadUint32(tm.lockCtx.Killed) != 0 {
 				return
 			}
-			bo := retry.NewBackofferWithVars(context.Background(), pessimisticLockMaxBackoff, c.txn.vars)
+			bo := retry.NewBackofferWithVars(context.Background(), keepAliveMaxBackoff, c.txn.vars)
```
Comment (youjiali1995) [inline store/tikv/2pc.go:749]: I think it shouldn't break the loop if the error is not a key error at L779.

File: store/tikv/pessimistic.go [hunk 2]
```diff
@@ -132,10 +132,22 @@ func (action actionPessimisticLock) handleSingleBatch(c *twoPhaseCommitter, bo *
 			return errors.Trace(err)
 		}
 		if regionErr != nil {
-			err = bo.Backoff(retry.BoRegionMiss, errors.New(regionErr.String()))
+			/// For other region error and the fake region error, backoff because
```
Comment (youjiali1995) [inline store/tikv/pessimistic.go:135]:
```suggestion
			// For other region error and the fake region error, backoff because
```

**Review by youjiali1995 — COMMENTED** [review]
Rest lgtm

**Review by youjiali1995 — APPROVED** [review]

File: store/tikv/2pc.go [hunk 3]
```diff
@@ -796,31 +808,36 @@ func sendTxnHeartBeat(bo *Backoffer, store *KVStore, primary []byte, startTS, tt
...
+				err = bo.Backoff(retry.BoRegionMiss, errors.New(regionErr.String()))
+				if err != nil {
+					return 0, errors.Trace(err), false
+				}
 			}
 			continue
 		}
 		if resp.Resp == nil {
-			return 0, errors.Trace(tikverr.ErrBodyMissing)
+			return 0, errors.Trace(err), false
```
Comment (lysu) [inline store/tikv/2pc.go:834]: those `err` should always be nil~?

**Review by lysu — COMMENTED** [review]
rest LGTM

File: store/tikv/2pc.go (hunk 3 above)
Comment (sticnarf) [reply store/tikv/2pc.go]: Sorry, it's a copy-paste mistake...

**Review by lysu — APPROVED** [review]

**Comment by bb7133** [comment]
```
[2021-06-05T07:53:19.089Z] GO111MODULE=on go build -o ../bin/revive github.com/mgechev/revive
[2021-06-05T07:53:19.089Z] linting
[2021-06-05T07:53:19.089Z]   ✘  https://revive.run/r#error-return  error should be the last type when returning multiple items  
[2021-06-05T07:53:19.089Z]   store/tikv/2pc.go:802:1
[2021-06-05T07:53:19.089Z] 
```
@sticnarf Please fix the linting check.

**Comment by bb7133** [comment]
https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_check/detail/tidb_ghpr_check/9030/pipeline

**Pull request merged**